crates/next-core/js/src/entry/router.ts

```diff
@@ -0,0 +1,148 @@
+import type { Ipc } from "@vercel/turbopack-next/ipc/index";
+import type { IncomingMessage } from "node:http";
+import { Buffer } from "node:buffer";
+import { createServer, makeRequest } from "@vercel/turbopack-next/ipc/server";
+import loadNextConfig from "@vercel/turbopack-next/entry/config/next";
+
+import "next/dist/server/node-polyfill-fetch.js";
+
+type RouterRequest = {
+  method: string;
+  pathname: string;
+  // TODO: not passed to request
+  headers: Record<string, string>;
+  query: Record<string, string>;
+};
+
+type RouteResult = {
+  url: string;
+  headers: Record<string, string>;
+  statusCode: number;
+  isRedirect: boolean;
+};
+
+type IpcOutgoingMessage =
+  | { type: "middleware-headers"; data: MiddlewareHeadersResponse }
+  | { type: "middleware-body"; data: Uint8Array }
+  | {
+      type: "redirect";
+      data: RedirectResponse;
+    }
+  | {
+      type: "rewrite";
+      data: RewriteResponse;
+    };
+
+type RedirectResponse = {
+  url: string;
+  statusCode: number;
+  headers: string[];
+};
+
+type RewriteResponse = {
+  url: string;
+  statusCode: number;
+  headers: string[];
+};
+
+type MiddlewareHeadersResponse = {
+  statusCode: number;
+  headers: string[];
+};
+
+export default async function route(
+  ipc: Ipc<RouterRequest, IpcOutgoingMessage>,
+  routerRequest: RouterRequest
+) {
+  // Deferring the import allows us to not error while we wait for Next.js to implement.
+  const { makeResolver } = await import("next/dist/server/router");
```

Comment:
```suggestion
  const { makeResolver } = await import("next/dist/server/router.js");
```
Looks like this needs to include extension.

![CleanShot 2023-01-25 at 15 13 00@2x](https://user-images.githubusercontent.com/22380829/214713047-8684f391-5a6c-4232-b231-f5ba06cbbad6.png)